File: hash-table/longest-palindrome.py
from collections import Counter

class Solution:
    def longestPalindrome(self, s: str) -> int:
        counter = Counter(s)
        res = 0
        is_odd = False
        for _, count in counter.items():
            if count % 2 == 0:
                res += count
            else:
                is_odd = True
                res += count - 1
        if is_odd:
            res += 1
        return res
        

File: hash-table/longest-palindrome.rb
# @param {String} s
# @return {Integer}
def longest_palindrome(s)
    res = 0
    odd_count_present = false
    s.chars.tally.values.each do |count|
        if count.even?
            res += count
        else
            res += count - 1
            odd_count_present = true
        end
    end
    res += 1 if odd_count_present
    res
end
